node/lib/stat_emitter.js

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var inherits = require('util').inherits;
+var EventEmitter = require('./event_emitter');
+var Stat = require('./stat');
+
+// This is an interface that exists only to explicitly declare the stats we are
+// tracking and avoid bugs due to mis-spelled names and types.
+
+function StatEmitter() {
```

Comment:
This entire thing seems to defeat the point of using events to loosely couple ourselves from `statsdClient`.

If we want to use the entire vocabulary of the `statsdClient` we should just pass it in and be done with it.

We may want to use MVP of something like this. 

```js
self.requestErrorsEvent = self.defineEvent('requestErrors');
self.requestsPendingEvent = self.defineEvent('requestsPending');

self.requestErrorsEvent.emit(new Stat.Counter(1, tags));
self.requestsPendingEvent.emit(new Stat.Gauge(self.pending));
```